vault/resource_gcp_auth_backend_role.go

```diff
@@ -3,13 +3,19 @@ package vault
 import (
 	"fmt"
 	"log"
+	"regexp"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 
 	"github.com/hashicorp/vault/api"
 )
 
+var (
+	gcpAuthBackendRoleBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/role/.+$")
```

Comment:
```suggestion
	gcpAuthBackendRoleBackendFromPathRegex = regexp.MustCompile("^auth/([^/]+)/role/.+$")
```

IIUC I think the current regex would return `gcp/role` as the backend for something like `auth/gcp/role/role/foo`. I don't think that's a meaningful path but it can't hurt to be a bit more specific about selecting strictly a single path component. Alternatively, you could use `.+?` to make the wildcard non-greedy and achieve the same result.

Similar for the regex below too.

Reply:
Oops I got it the wrong way round. LGTM 👍